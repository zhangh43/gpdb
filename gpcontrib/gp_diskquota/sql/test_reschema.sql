-- Test re-set_schema_quota
CREATE SCHEMA srE;
SELECT diskquota.set_schema_quota('srE', '1 MB');
SET search_path TO srE;
CREATE TABLE a(i int);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,1000000000);
-- expect insert fail when exceed quota limit
INSERT INTO a SELECT generate_series(1,1000);
-- set schema quota larger
SELECT diskquota.set_schema_quota('srE', '1 GB');
SELECT pg_sleep(5);
-- expect insert succeed
INSERT INTO a SELECT generate_series(1,1000);

DROP TABLE a;
RESET search_path;
DROP SCHEMA srE;

