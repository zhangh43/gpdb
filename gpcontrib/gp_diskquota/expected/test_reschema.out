-- Test re-set_schema_quota
create schema srE;
select diskquota.set_schema_quota('srE', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

set search_path to srE;
create table a(i int);
-- expect insert fail
insert into a select generate_series(1,1000000000);
ERROR:  schema's disk space quota exceeded with name:sre
-- expect insert fail when exceed quota limit
insert into a select generate_series(1,1000);
ERROR:  schema's disk space quota exceeded with name:sre
-- set schema quota larger
select diskquota.set_schema_quota('srE', '1 GB');
 set_schema_quota 
------------------
 
(1 row)

select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert succeed
insert into a select generate_series(1,1000);
drop table a;
reset search_path;
drop schema srE;
